laplace/utils/matrix.py

```diff
@@ -432,11 +432,46 @@ def bmm(self, W: torch.Tensor, exponent: float = -1) -> torch.Tensor:
         else:
             raise ValueError('Invalid shape for W')
 
+    def diag(self, exponent: float = 1) -> torch.Tensor:
+        """Extract diagonal of the entire decomposed Kronecker factorization.
+
+        Parameters
+        ----------
+        exponent: float, default=1
+            exponent of the Kronecker factorization
+
+        Returns
+        -------
+        diag : torch.Tensor
+        """
+        diags = list()
+        for Qs, ls, delta in zip(self.eigenvectors, self.eigenvalues, self.deltas):
+            if len(ls) == 1:
+                Ql = Qs[0] * torch.pow(ls[0] + delta, exponent).reshape(1, -1)
+                d = torch.einsum('mp,mp->m', Ql, Qs[0])  # only compute inner products for diag
+                diags.append(d)
+            else:  
+                Q1, Q2 = Qs
+                l1, l2 = ls
+                if self.damping:
+                    delta_sqrt = torch.sqrt(delta)
+                    l = torch.pow(torch.ger(l1 + delta_sqrt, l2 + delta_sqrt), exponent)
```

Comment:
Maybe replace `torch.ger` with `torch.outer`; it is more readable and `torch.ger` [is deprecated](https://pytorch.org/docs/stable/generated/torch.ger.html).